engine/src/Interpreter/interpreter_ops.cuh

```diff
@@ -152,512 +114,264 @@ public:
-			cudf::mutable_column_view & c = out_table.column(i);
-			host_out_data_ptrs[i] = c.data<void>();
-			host_out_valid_ptrs[i] = c.null_mask();
-		}
-
-		CUDA_TRY(cudaMemcpyAsync(this->output_data,
-			host_out_data_ptrs.data(),
-			sizeof(void *) * num_final_outputs,
-			cudaMemcpyHostToDevice,
-			stream));
-		CUDA_TRY(cudaMemcpyAsync(this->valid_ptrs_out,
-			host_out_valid_ptrs.data(),
-			sizeof(cudf::bitmask_type *) * num_final_outputs,
-			cudaMemcpyHostToDevice,
-			stream));
-
-
-		// copy over inputs
-		std::vector<cudf::type_id> left_input_types_vec(num_operations);
-		std::vector<cudf::type_id> right_input_types_vec(num_operations);
-		std::vector<cudf::type_id> output_types_vec(num_operations);
-		std::vector<cudf::type_id> output_final_types_vec(num_final_outputs);
-		std::vector<int64_t> left_scalars_host(num_operations);
-		std::vector<int64_t> right_scalars_host(num_operations);
-		std::map<int, cudf::type_id> output_map_type;
-		for(int i = 0; i < num_operations; i++) {
-			column_index_type left_index = left_input_positions_vec[i];
-			column_index_type right_index = right_input_positions_vec[i];
-			column_index_type output_index = output_positions_vec[i];
-
-			if(left_index >= 0 && left_index < num_columns) {
-				left_input_types_vec[i] = table.column(left_index).type().id();
-			} else if(left_index == SCALAR_NULL_INDEX) {
-				left_input_types_vec[i] = cudf::type_id::EMPTY;
-			} else if(left_index == SCALAR_INDEX) {
-				left_input_types_vec[i] = left_scalars[i]->type().id();
-				left_scalars_host[i] = cudf::experimental::type_dispatcher(
-					left_scalars[i]->type(), upcast_to_64_bit_functor{}, left_scalars[i].get());
-			} else if(left_index == UNARY_INDEX) {
-					// wont be used its a unary operation
-			} else {
-				// have to get it from the output that generated it
-				left_input_types_vec[i] = output_map_type[left_index];
-			}
-
-			if(right_index >= 0 && right_index < num_columns) {
-				right_input_types_vec[i] = table.column(right_index).type().id();
-			} else if(right_index == SCALAR_NULL_INDEX) {
-				right_input_types_vec[i] = cudf::type_id::EMPTY;
-			} else if(right_index == SCALAR_INDEX) {
-				right_input_types_vec[i] = right_scalars[i]->type().id();
-				right_scalars_host[i] = cudf::experimental::type_dispatcher(
-					right_scalars[i]->type(), upcast_to_64_bit_functor{}, right_scalars[i].get());
-			} else if(right_index == UNARY_INDEX) {
-					// wont be used its a unary operation
-			} else {
-				// have to get it from the output that generated it
-				right_input_types_vec[i] = output_map_type[right_index];
-			}
-
-			cudf::type_id type_from_op;
-			if(right_index == UNARY_INDEX) {
-				// unary
-				type_from_op = get_output_type(left_input_types_vec[i], unary_operators[i]);
-			} else {
-				// binary
-				type_from_op = get_output_type(left_input_types_vec[i], right_input_types_vec[i], operators[i]);
-			}
+		void * temp_valids_out_buffer)
+		: out_table{out_table},
+			table{table},
```

Comment:
Not anymore since tables and scalars are passed in a different way I ended up deleting most of the data in `temp_space`  buffer. For now I created `rmm::device_vector`s for the remaining data and I pass the pointer to the functor, when the migration is completed we could run some benchmarks and go back to one unique buffer for the remaining data if that's the case